pkg/k8s/service.go

```diff
@@ -156,6 +164,7 @@ func ParseService(svc *slim_corev1.Service, nodeAddressing datapath.NodeAddressi
 					}
 				}
 				if option.Config.EnableIPv6 &&
+					couldBeProtocol(svc.Spec.IPFamily, slim_corev1.IPv6Protocol) &&
```

Comment:
if svc.Spec.IPFamily is set as IPv4 explicitly, is it expected behavior to set ipv6 value ?

Reply:
The issue here is that "IPFamily" could be an empty string. Empty string and "IPv6" are the two valid options.

Reply:
yes, it was related to the below point as couldBeProtocol was allowing IPv4 explicitly.